packages/protocol/contracts/L1/IProofVerifier.sol

```diff
@@ -0,0 +1,25 @@
+// SPDX-License-Identifier: MIT
+//  _____     _ _         _         _
+// |_   _|_ _(_) |_____  | |   __ _| |__ ___
+//   | |/ _` | | / / _ \ | |__/ _` | '_ (_-<
+//   |_|\__,_|_|_\_\___/ |____\__,_|_.__/__/
+
+pragma solidity ^0.8.18;
+
+import {TaikoData} from "./TaikoData.sol";
+import {AddressResolver} from "../common/AddressResolver.sol";
+
+interface IProofVerifier {
+    /**
+     * Verifying proof via the ProofVerifier contract
+     *
+     * @param instance Hashed public input
+     * @param blockProofs Proof array
+     * @param resolver Current (up-to-date) address resolver
+     */
+    function verifyProofs(
+        bytes32 instance, 
+        TaikoData.TypedProof[] calldata blockProofs,
+        AddressResolver resolver
+    ) external;
```

Comment:
```suggestion
    function verifyProof(bytes calldata proof) external;
```

What do you think of this API? This way we don't have any dependencies and the proof can really be whatever, and proofs don't have to confirm to `TypedProof` which may not always fit.